tardis/workflows/standard_simulation_solver.py

```diff
@@ -0,0 +1,318 @@
+                )
+
+        self.simulation_state.t_radiative = next_values["t_radiative"]
+        self.simulation_state.dilution_factor = next_values["dilution_factor"]
+        self.simulation_state.blackbody_packet_source.temperature = next_values[
+            "t_inner"
+        ]
+
+    def solve_plasma(
+        self,
+        transport_state,
+    ):
+        update_properties = dict(
+            t_rad=self.simulation_state.t_radiative,
+            w=self.simulation_state.dilution_factor,
+        )
+        # A check to see if the plasma is set with JBluesDetailed, in which
+        # case it needs some extra kwargs.
+        if "j_blue_estimator" in self.plasma_solver.outputs_dict:
+            update_properties.update(
+                t_inner=self.simulation_state.blackbody_packet_source.temperature,
+                j_blue_estimator=transport_state.radfield_mc_estimators.j_blue_estimator,
+            )
+
+        self.plasma_solver.update(**update_properties)
+
+    def solve_montecarlo(self, no_of_real_packets, no_of_virtual_packets=0):
+        transport_state = self.transport_solver.initialize_transport_state(
+            self.simulation_state,
+            self.plasma_solver,
+            no_of_real_packets,
+            no_of_virtual_packets=no_of_virtual_packets,
+            iteration=self.completed_iterations,
+        )
+
+        virtual_packet_energies = self.transport_solver.run(
+            transport_state,
+            time_explosion=self.simulation_state.time_explosion,
+            iteration=self.completed_iterations,
+            total_iterations=self.total_iterations,
+            show_progress_bars=False,
+        )
+
+        output_energy = transport_state.packet_collection.output_energies
+        if np.sum(output_energy < 0) == len(output_energy):
+            logger.critical("No r-packet escaped through the outer boundary.")
+
+        return transport_state, virtual_packet_energies
+
+    def initialize_spectrum_solver(
+        self,
+        transport_state,
+        virtual_packet_energies=None,
+    ):
+        # Set up spectrum solver
+        self.spectrum_solver.transport_state = transport_state
+
+        if virtual_packet_energies is not None:
+            self.spectrum_solver._montecarlo_virtual_luminosity.value[
+                :
+            ] = virtual_packet_energies
+
+        if self.integrated_spectrum_settings is not None:
+            # Set up spectrum solver integrator
+            self.spectrum_solver.integrator_settings = (
+                self.integrated_spectrum_settings
+            )
+            self.spectrum_solver._integrator = FormalIntegrator(
+                self.simulation_state, self.plasma_solver, self.transport_solver
+            )
+
+    def solve(self):
+        converged = False
+        while self.completed_iterations < self.total_iterations - 1:
+            transport_state, virtual_packet_energies = self.solve_montecarlo(
```

Comment:
virtual packet stuff should not be coming out of the montecarlo solver in the future